src/ui/public/kuery/functions/utils/get_fields_by_wildcard.js

```diff
@@ -0,0 +1,43 @@
+import _ from 'lodash';
+
+// Copied from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
+function escapeRegExp(string) {
+  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
+}
+
+/**
+ * Returns a list of fields matching the given pattern.
+ * @param {string} pattern - Field name with optional wildcards.
+ * @param indexPattern
+ * @return {Field[]} - a list of index pattern Fields
+ */
+export function getFieldsByWildcard(pattern, indexPattern) {
+  if (pattern.includes('*')) {
```

Comment:
I wasn't planning on adding it in this PR. Do you feel strongly that it should be? I was going to wait and see how much demand there is since it's always easier to add things than take them away down the road.